exercises/react/src/example.c

```diff
@@ -0,0 +1,211 @@
+#include <stdlib.h>
+#include "react.h"
+
+enum cell_kind {
+   kind_input,
+   kind_compute1,
+   kind_compute2,
+};
+
+struct child {
+   struct cell *cell;
+   struct child *next;
+};
+
+struct cb {
+   callback_id id;
+   callback f;
+   void *obj;
+   struct cb *next;
+};
+
+struct reactor {
+   struct child *first;
+   struct child *last;
+};
+
+struct cell {
+   struct reactor *reactor;
+   int value;
+   enum cell_kind kind;
+   struct cell *input1;
+   struct cell *input2;
```

Comment:
Personally, I'm okay with casting since it's all internal to this module, but I don't think the extra complexity is warranted. Let's leave it as-is to keep the solution more approachable.